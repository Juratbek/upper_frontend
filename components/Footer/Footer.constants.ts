import { IFooterLink } from './Footer.types';

export const LINKS: IFooterLink[] = [
  {
    text: "Qo'llab quvvatlash",
    url: '/sponsor',
  },
  {
    text: 'Aloqa',
    url: '/contacts',
  },
  {
    text: "Qo'llanma",
    url: '/docs/write-article_introduction_quick-start',
  },
  {
    text: 'Biz haqimizda',
    url: '/about-us',
  },
  {
    text: 'Jamoa',
    url: '/team',
  },
];
