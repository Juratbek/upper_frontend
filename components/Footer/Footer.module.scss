@import 'styles/mixins';

.root {
  height: 12rem;
  width: 100%;
}

.container {
  display: flex;
  align-items: center;
  height: 100%;
  border-top: 1px solid var(--black-50);
}

.logo svg {
  width: 8rem;
  height: 3rem;
  margin: 0 auto;
}

.links {
  display: flex;
  gap: 4rem;
  padding: 0;

  @include screenMax576 {
    flex-wrap: wrap;
    gap: 5%;
  }

  & > li {
    list-style: none;

    @include screenMax576 {
      width: 45%;
      margin-bottom: 1rem;
    }
  }
}

.media-icons {
  display: flex;
  gap: 1.6rem;
  padding: 0;

  .media-icon-btn {
    list-style: none;
    width: 4rem;
    height: 4rem;
    border-radius: 100rem;
    background: var(--blue-500);
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

.text {
  color: var(--black-400);
  font-style: normal;
  font-weight: 400;
}

