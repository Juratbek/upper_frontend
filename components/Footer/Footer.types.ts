export interface IFooterLink {
  text: string;
  url: string;
}
