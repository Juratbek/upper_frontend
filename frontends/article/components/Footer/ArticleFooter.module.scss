.root {
  display: flex;
}

.reactions-container {
  display: flex;
  align-items: center;
  padding: 1rem 2rem;
  border-radius: 10rem;
  background: var(--card-bg);

  .like-count {
    margin-left: 1.2rem;
  }

  .divider {
    $margin: 1.6rem;
    margin-left: $margin;
    margin-right: $margin;
    height: 2rem;
  }
}

.count {
  color: var(--comments-count);
  font-size: 1.4rem;
  font-weight: 500;
  line-height: 1.4rem;
}

.comment-container {
  display: flex;
  padding: 10px 16px;
  align-items: center;
  gap: 12px;
  border-radius: 300px;
  background: var(--card-bg);
  margin-left: 12px;
  border: none;
  cursor: pointer;
}

.actions-container {
  margin-left: auto;
  display: flex;
  gap: 12px;
  position: relative;
}