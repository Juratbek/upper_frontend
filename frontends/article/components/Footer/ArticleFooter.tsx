import { CommentIcon, DislikeIcon, LikeIcon, ShareIcon } from 'components/icons';
import { Clickable, Divider } from 'components/lib';
import { SharePopover } from 'components/organisms';
import { UPPER_BLUE_COLOR } from 'constants/colors';
import { useAuth, useClickOutside, useTheme } from 'hooks';
import { useRouter } from 'next/router';
import { FC, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useCommentsCount } from 'store/clients/comments';
import { ReactionType, useHasReaction, useReact, useReactionCount } from 'store/clients/reaction';
import { toggleCommentsModal } from 'store/states';
import { getClassName } from 'utils';

import classes from './ArticleFooter.module.scss';

export const ArticleFooter: FC<{ sharePopoverId: string }> = ({ sharePopoverId }) => {
  const { query } = useRouter();
  const articleId = Number(query.id);
  const dispatch = useDispatch();
  const { isAuthenticated, openLoginPage } = useAuth();
  const { data: likeCountData, refetch: refetchLikeCount } = useReactionCount(
    articleId,
    ReactionType.LIKE,
  );

  const { data: likeData, refetch: refetchIsLiked } = useHasReaction({
    articleId,
    isAuthenticated,
    type: ReactionType.LIKE,
  });
  const { data: dislikeData, refetch: refetchIsDisliked } = useHasReaction({
    articleId,
    isAuthenticated,
    type: ReactionType.DISLIKE,
  });

  const refetchLikeDislikeQueries = () => {
    refetchLikeCount();
    refetchIsLiked();
    refetchIsDisliked();
  };

  const { mutate: like, isPending: isLikePending } = useReact(
    articleId,
    ReactionType.LIKE,
    refetchLikeDislikeQueries,
  );
  const { mutate: dislike, isPending: isDisLikePending } = useReact(
    articleId,
    ReactionType.DISLIKE,
    refetchLikeDislikeQueries,
  );

  const { data: commentsCountData } = useCommentsCount(articleId);
  const [isOpen, setIsOpen] = useState(false);
  const { themeColors } = useTheme();

  const commentClickHandler = (): unknown => dispatch(toggleCommentsModal());
  const shareClickHandler = (): unknown => setIsOpen((prev) => !prev);
  const closeSharePopover = (): unknown => setIsOpen(false);

  const [rootRef] = useClickOutside(closeSharePopover, `#${sharePopoverId}`);

  const likeHandler = (): void => {
    if (isAuthenticated) {
      like();
    } else {
      openLoginPage("Iltimos maqolaga layk bosish uchun ro'yxatdan o'ting");
    }
  };

  const dislikeHandler = (): void => {
    if (isAuthenticated) {
      dislike();
    } else {
      openLoginPage("Iltimos maqolaga dislayk bosish uchun ro'yxatdan o'ting");
    }
  };

  return (
    <div className={classes.root}>
      <div className={classes['reactions-container']}>
        <Clickable
          loading={isLikePending}
          disabled={likeData?.hasReaction || isLikePending}
          onClick={likeHandler}
          data-testid='like-icon'
        >
          <LikeIcon color={likeData?.hasReaction ? UPPER_BLUE_COLOR : themeColors.icon} />
        </Clickable>
        {Boolean(Number(likeCountData?.count)) && (
          <span className={getClassName(classes['like-count'], classes.count)}>
            {likeCountData!.count}
          </span>
        )}
        <Divider color='secondary' className={classes.divider} type='vertical' />
        <Clickable
          loading={isDisLikePending}
          disabled={dislikeData?.hasReaction || isDisLikePending}
          onClick={dislikeHandler}
        >
          <DislikeIcon color={dislikeData?.hasReaction ? UPPER_BLUE_COLOR : themeColors.icon} />
        </Clickable>
      </div>
      <Clickable
        onClick={commentClickHandler}
        data-testid='comment-icon'
        className={classes['comment-container']}
      >
        <CommentIcon color={themeColors.icon} />
        {Boolean(Number(commentsCountData?.count)) && (
          <span className={classes['count']}>{commentsCountData!.count}</span>
        )}
      </Clickable>
      <div className={classes['actions-container']}>
        {/* <Clickable>
          <SaveIcon />
        </Clickable> */}
        <Clickable
          className={classes.share}
          data-testid='share-icon'
          onClick={shareClickHandler}
          id={sharePopoverId}
        >
          <ShareIcon color={themeColors.icon} />
        </Clickable>
        <SharePopover ref={rootRef} isOpen={isOpen} closeSharePopover={closeSharePopover} />
      </div>
    </div>
  );
};
