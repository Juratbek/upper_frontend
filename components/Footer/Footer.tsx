import Link from 'next/link';
import { FC } from 'react';

import { LINKS } from './Footer.constants';
import styles from './Footer.module.scss';

export const Footer: FC = () => {
  return (
    <div className={styles.footerContainer}>
      <ul className={styles.footerNav}>
        {LINKS.map((link) => (
          <li key={link.text}>
            <Link href={link.url}>
              <a>{link.text}</a>
            </Link>
          </li>
        ))}
      </ul>
      <p className={styles.copyrightText}>
        Â© {new Date().getFullYear()} UPPER | Barcha huquqlar himoyalangan
      </p>
    </div>
  );
};
