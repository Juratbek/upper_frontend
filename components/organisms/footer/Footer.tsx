import { Logo } from 'components/molecules';
import Link from 'next/link';
import { FC } from 'react';
import { WEB_APP_ROOT_DIR } from 'variables';

import { LINKS, MEDIA_ICONS } from './Footer.constants';
import styles from './Footer.module.scss';

export const Footer: FC = () => {
  return (
    <footer className={styles.root}>
      <div className={`${styles.container} container`}>
        <div className={styles['main-block']}>
          <Logo />
          <ul className={styles.links}>
            {LINKS.map((link) => (
              <li key={link.text} className={styles.text}>
                <Link href={WEB_APP_ROOT_DIR + link.url}>{link.text}</Link>
              </li>
            ))}
          </ul>
        </div>
        <div className={styles['secondary-block']}>
          <p className={styles.text}>
            Â© {new Date().getFullYear()} UPPER | Barcha huquqlar himoyalangan
          </p>
          <ul className={styles['media-icons']}>
            {MEDIA_ICONS.map(({ icon: Icon, url }) => (
              <li key={Icon.toString()} className={styles['media-icon-btn']}>
                <Link href={url} target='_blank'>
                  <Icon width={20} height={20} color='white' />
                </Link>
              </li>
            ))}
          </ul>
        </div>
      </div>
    </footer>
  );
};
