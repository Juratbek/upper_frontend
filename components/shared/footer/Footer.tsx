import { useTheme } from 'hooks';
import Link from 'next/link';
import { FC } from 'react';
import { ICONS, WEB_APP_ROOT_DIR } from 'variables';

import { LINKS, MEDIA_ICONS } from './Footer.constants';
import styles from './Footer.module.scss';

const Logo = ICONS.logo;

export const Footer: FC = () => {
  const { themeColors } = useTheme();

  return (
    <div className={styles.root}>
      <div className={`${styles.container} container`}>
        <span className={styles.logo}>
          <Logo color={themeColors.icon} />
        </span>
        <ul className={styles.links}>
          {LINKS.map((link) => (
            <li key={link.text} className={styles.text}>
              <Link href={WEB_APP_ROOT_DIR + link.url}>{link.text}</Link>
            </li>
          ))}
        </ul>
        <p className={styles.text}>
          Â© {new Date().getFullYear()} UPPER | Barcha huquqlar himoyalangan
        </p>
        <ul className={styles['media-icons']}>
          {MEDIA_ICONS.map(({ icon: Icon, url }) => (
            <li key={Icon.toString()} className={styles['media-icon-btn']}>
              <Link href={url} target='_blank'>
                <Icon width={20} height={20} color='white' />
              </Link>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};
