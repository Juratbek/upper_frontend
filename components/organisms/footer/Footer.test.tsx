import { render, screen } from '@testing-library/react';

import { Footer } from './Footer';

describe('Footer', () => {
  it('renders the Footer component', () => {
    render(<Footer />);
    const footer = screen.getByRole('contentinfo');

    expect(footer).toMatchSnapshot();
  });
});
