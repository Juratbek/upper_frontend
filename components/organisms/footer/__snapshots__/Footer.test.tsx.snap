// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > renders the Footer component 1`] = `
<footer
  class="root"
>
  <div
    class="container container"
  >
    <div
      class="main-block"
    >
      <a
        class="container link"
        data-testid="logo-link"
        href="/web"
      >
        <svg
          data-testid="logo-svg"
          fill="none"
          role="img"
          viewBox="0 0 188 80"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            fill="white"
            id="path-1-inside-1_804_345"
          >
            <path
              d="M0 0H80V80H0V0Z"
            />
          </mask>
          <path
            d="M0 0V-5H-5V0H0ZM0 80H-5V85H0V80ZM0 5H80V-5H0V5ZM80 75H0V85H80V75ZM5 80V0H-5V80H5Z"
            fill="#54A9EB"
            mask="url(#path-1-inside-1_804_345)"
          />
          <line
            stroke="#54A9EB"
            stroke-width="5"
            x1="77.5"
            x2="77.5"
            y1="5"
            y2="17"
          />
          <line
            stroke="#54A9EB"
            stroke-width="5"
            x1="77.5"
            x2="77.5"
            y1="63"
            y2="75"
          />
          <path
            d="M50.8324 21.9091H55.3239V46.4673C55.3239 49.0028 54.7262 51.2667 53.5309 53.2589C52.3477 55.239 50.6754 56.8026 48.5142 57.9496C46.353 59.0845 43.8175 59.652 40.9077 59.652C37.9979 59.652 35.4624 59.0845 33.3011 57.9496C31.1399 56.8026 29.4616 55.239 28.2663 53.2589C27.0831 51.2667 26.4915 49.0028 26.4915 46.4673V21.9091H30.983V46.1051C30.983 47.9162 31.3814 49.5281 32.1783 50.9407C32.9751 52.3413 34.1101 53.446 35.5831 54.255C37.0682 55.0518 38.843 55.4503 40.9077 55.4503C42.9723 55.4503 44.7472 55.0518 46.2322 54.255C47.7173 53.446 48.8523 52.3413 49.6371 50.9407C50.434 49.5281 50.8324 47.9162 50.8324 46.1051V21.9091Z"
            fill="#54A9EB"
          />
          <path
            d="M64.2932 59V21.9091H76.8259C79.7357 21.9091 82.1143 22.4343 83.9616 23.4847C85.8209 24.5231 87.1974 25.9297 88.0908 27.7045C88.9843 29.4794 89.431 31.4595 89.431 33.6449C89.431 35.8303 88.9843 37.8164 88.0908 39.6033C87.2094 41.3903 85.8451 42.815 83.9978 43.8775C82.1505 44.9279 79.784 45.4531 76.8983 45.4531H67.9154V41.4688H76.7535C78.7457 41.4688 80.3454 41.1246 81.5528 40.4364C82.7602 39.7482 83.6356 38.8185 84.1789 37.6474C84.7343 36.4641 85.012 35.13 85.012 33.6449C85.012 32.1598 84.7343 30.8317 84.1789 29.6605C83.6356 28.4893 82.7542 27.5717 81.5347 26.9077C80.3153 26.2315 78.6974 25.8935 76.681 25.8935H68.7847V59H64.2932ZM96.6663 59V21.9091H109.199C112.109 21.9091 114.487 22.4343 116.335 23.4847C118.194 24.5231 119.57 25.9297 120.464 27.7045C121.357 29.4794 121.804 31.4595 121.804 33.6449C121.804 35.8303 121.357 37.8164 120.464 39.6033C119.582 41.3903 118.218 42.815 116.371 43.8775C114.524 44.9279 112.157 45.4531 109.271 45.4531H100.288V41.4688H109.127C111.119 41.4688 112.718 41.1246 113.926 40.4364C115.133 39.7482 116.009 38.8185 116.552 37.6474C117.107 36.4641 117.385 35.13 117.385 33.6449C117.385 32.1598 117.107 30.8317 116.552 29.6605C116.009 28.4893 115.127 27.5717 113.908 26.9077C112.688 26.2315 111.07 25.8935 109.054 25.8935H101.158V59H96.6663ZM129.039 59V21.9091H151.424V25.8935H133.531V38.4261H150.265V42.4105H133.531V55.0156H151.714V59H129.039ZM159.52 59V21.9091H172.052C174.95 21.9091 177.329 22.4041 179.188 23.3942C181.047 24.3722 182.424 25.7184 183.317 27.4329C184.211 29.1474 184.658 31.0973 184.658 33.2827C184.658 35.468 184.211 37.4059 183.317 39.0962C182.424 40.7866 181.054 42.1147 179.206 43.0806C177.359 44.0344 174.998 44.5114 172.125 44.5114H161.983V40.4545H171.98C173.96 40.4545 175.554 40.1648 176.761 39.5852C177.981 39.0057 178.862 38.1847 179.405 37.1222C179.961 36.0476 180.239 34.7678 180.239 33.2827C180.239 31.7976 179.961 30.4996 179.405 29.3888C178.85 28.2781 177.963 27.4208 176.743 26.8171C175.524 26.2013 173.912 25.8935 171.908 25.8935H164.011V59H159.52ZM176.979 42.3381L186.106 59H180.891L171.908 42.3381H176.979Z"
            fill="black"
          />
        </svg>
      </a>
      <ul
        class="links"
      >
        <li
          class="text"
        >
          <a
            href="https://github.com/Juratbek/upper_frontend/blob/beta/CONTRIBUTING.md"
            target="_blank"
          >
            Sayt faoliyatiga hissa qo'shish
          </a>
        </li>
        <li
          class="text"
        >
          <a
            href="/web/contacts"
          >
            Aloqa
          </a>
        </li>
        <li
          class="text"
        >
          <a
            href="/web/team"
          >
            Jamoa
          </a>
        </li>
      </ul>
    </div>
    <div
      class="secondary-block"
    >
      <p
        class="text"
      >
        Â© 
        2024
         UPPER | Barcha huquqlar himoyalangan
      </p>
      <ul
        class="media-icons"
      >
        <li
          class="media-icon-btn"
        >
          <a
            href="https://t.me/upperuz_bot"
            target="_blank"
          >
            <svg
              fill="none"
              height="20"
              role="icon"
              viewBox="0 0 24 24"
              width="20"
            >
              <path
                d="M4.80626 9.87148L16.7519 5.25521C18.1945 4.6977 18.9159 4.41895 19.3657 4.78324C19.8156 5.14754 19.6929 5.91106 19.4474 7.43812L17.9078 17.0181C17.6912 18.3656 17.583 19.0394 17.0937 19.2612C16.6044 19.4829 16.0264 19.1202 14.8702 18.3949L13.3248 17.4252C11.3333 16.1758 10.3375 15.551 10.2362 14.5825C10.1348 13.614 10.9797 12.7967 12.6694 11.162L15.9697 7.96934C16.1366 7.80791 16.0223 7.52542 15.7901 7.52542C15.7406 7.52542 15.6921 7.53965 15.6505 7.56641L7.70166 12.6736C7.18785 13.0037 6.93094 13.1688 6.64783 13.2498C6.48484 13.2964 6.31657 13.3221 6.14709 13.3263C5.85271 13.3336 5.55821 13.2529 4.96922 13.0914C3.30827 12.6359 2.4778 12.4082 2.28331 11.984C2.17665 11.7513 2.16325 11.4866 2.24587 11.2444C2.39654 10.8027 3.19978 10.4923 4.80626 9.87148Z"
                fill="white"
              />
            </svg>
          </a>
        </li>
        <li
          class="media-icon-btn"
        >
          <a
            href="https://linkedin"
            target="_blank"
          >
            <svg
              fill="none"
              height="20"
              role="icon"
              viewBox="0 0 24 24"
              width="20"
            >
              <path
                d="M5.58789 7.22266C4.5918 7.22266 3.82031 6.44141 3.82031 5.51367C3.82031 4.58594 4.5918 3.81445 5.58789 3.81445C6.59375 3.81445 7.35547 4.58594 7.35547 5.51367C7.35547 6.44141 6.59375 7.22266 5.58789 7.22266ZM3.85938 19V8.36523H7.31641V19H3.85938ZM9.00586 19V8.36523H12.4727V10.2402H12.541C13.0684 8.90234 14.1621 8.14062 15.7734 8.14062C18.1074 8.14062 19.4258 9.63477 19.4258 12.1543V19H15.9688V12.8965C15.9688 11.6758 15.3926 10.9629 14.2598 10.9629C13.1562 10.9629 12.4727 11.7832 12.4727 12.9648V19H9.00586Z"
                fill="white"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
  </div>
</footer>
`;
